### PR TITLE
Support multiple XRPL door addresses

### PR DESCRIPTION
**Pull request by surangap**
### Changelog
 - Adds support for multiple XRPL door addresses in the `pallet-xrpl-bridge`
 - storage items `DoorTicketSequence, DoorTicketSequenceParams, DoorTicketSequenceParamsNext, TicketSequenceThresholdReachedEmitted, DoorTxFee, DoorAddress` has been changed to `StorageMap`s
 - `XRPLDoorAccount` enum is added to represent different door accounts supported.
 - `EthyToXrplBridgeAdapter` and ethy has been updated to support multiple XRPL door accounts.

### CODE REVIEW

File: pallet/xrpl-bridge/src/lib.rs [hunk 1]
```diff
@@ -375,31 +387,43 @@ pub mod pallet {
...
 	#[pallet::storage]
 	#[pallet::getter(fn door_ticket_sequence_params_next)]
-	/// The Ticket sequence params of the XRPL door account for the next allocation
+	/// The Ticket sequence params of the XRPL door accounts for the next allocation
 	pub type DoorTicketSequenceParamsNext<T: Config> =
-		StorageValue<_, XrplTicketSequenceParams, ValueQuery>;
+		StorageMap<_, Twox64Concat, XRPLDoorAccount, XrplTicketSequenceParams, ValueQuery>;
 
 	#[pallet::type_value]
 	pub fn DefaultTicketSequenceThresholdReachedEmitted() -> bool {
```
Comment (JasonTulp) [inline pallet/xrpl-bridge/src/lib.rs:413]: This code is redundant

Comment (surangap) [reply pallet/xrpl-bridge/src/lib.rs]: [15827b1](https://github.com/futureversecom/trn-seed/pull/884/commits/15827b1400dc20ea6e33de8ac76d1f2472ff2408)

File: pallet/ethy/src/impls.rs [hunk 2]
```diff
@@ -891,15 +891,16 @@ impl<T: Config> Pallet<T> {
...
+				// Any observer can subscribe to this event and submit the resulting proofs to keep
+				// the authority set of the xrpl door addresses updated.
 				Self::deposit_event(Event::<T>::XrplAuthoritySetChange {
-					event_proof_id,
+					event_proof_ids: event_proof_ids.clone(),
 					validator_set_id: next_validator_set_id,
 				});
-				XrplNotarySetProofId::<T>::put(event_proof_id);
+				// NOTE - keep the first event proof id for the sake of display. can be removed later if no other use
+				XrplNotarySetProofId::<T>::put(event_proof_ids[0]);
```
Comment (JasonTulp) [inline pallet/ethy/src/impls.rs:903]: nit, moving the storage put above the event will avoid an unnecessary clone

File: pallet/ethy/src/lib.rs [hunk 3]
```diff
@@ -422,8 +422,8 @@ pub mod pallet {
 		/// A proof for the change will be generated with the given `event_id`
 		AuthoritySetChange { event_proof_id: EventProofId, validator_set_id: u64 },
 		/// A notary (validator) set change for Xrpl is in motion
-		/// A proof for the change will be generated with the given `event_id`
-		XrplAuthoritySetChange { event_proof_id: EventProofId, validator_set_id: u64 },
+		/// A proofs for the change will be generated with the given `event_proof_ids`
```
Comment (JasonTulp) [inline pallet/ethy/src/lib.rs:425]:
```suggestion
		/// Some proofs for the change will be generated with the given `event_proof_ids`
```

File: pallet/xrpl-bridge/src/lib.rs [hunk 4]
```diff
@@ -615,17 +634,26 @@ pub mod pallet {
...
-		/// Set XRPL door address managed by this pallet
+		/// Set/Reset XRPL door addresses managed by this pallet
 		#[pallet::call_index(9)]
 		#[pallet::weight((T::WeightInfo::set_door_address(), DispatchClass::Operational))]
 		#[transactional]
 		pub fn set_door_address(
 			origin: OriginFor<T>,
-			door_address: XrplAccountId,
+			door_account: XRPLDoorAccount,
+			door_address: Option<XrplAccountId>,
```
Comment (JasonTulp) [inline pallet/xrpl-bridge/src/lib.rs:644]: Why add the option to remove the door_address?

File: pallet/xrpl-bridge/src/lib.rs [hunk 5]
```diff
@@ -634,27 +662,32 @@ pub mod pallet {
...
 				|| start_ticket_sequence < current_params.start_sequence
 				|| ticket_bucket_size == 0
 			{
 				fail!(Error::<T>::NextTicketSequenceParamsInvalid);
 			}
-			DoorTicketSequenceParamsNext::<T>::put(XrplTicketSequenceParams {
-				start_sequence: start_ticket_sequence,
-				bucket_size: ticket_bucket_size,
-			});
+			DoorTicketSequenceParamsNext::<T>::set(
```
Comment (JasonTulp) [inline pallet/xrpl-bridge/src/lib.rs:682]: Why use set instead of insert?

**Review by JasonTulp — APPROVED** [review]
lgtm!

**Review by KarishmaBothara — APPROVED** [review]
lgtm

File: pallet/xrpl-bridge/src/lib.rs (hunk 4 above)
Comment (surangap) [reply pallet/xrpl-bridge/src/lib.rs:644]: Now that door addresses are stored in a map, we could use the same extrinsic to set/unset the entries.

File: pallet/ethy/src/impls.rs (hunk 2 above)
Comment (surangap) [reply pallet/ethy/src/impls.rs]: [68dbbfb](https://github.com/futureversecom/trn-seed/pull/884/commits/68dbbfb8c9209a03b2ff03f5b7b3e495bcc57d25)

File: pallet/ethy/src/lib.rs (hunk 3 above)
Comment (surangap) [reply pallet/ethy/src/lib.rs]: [68dbbfb](https://github.com/futureversecom/trn-seed/pull/884/commits/68dbbfb8c9209a03b2ff03f5b7b3e495bcc57d25)

File: pallet/xrpl-bridge/src/lib.rs (hunk 5 above)
Comment (surangap) [reply pallet/xrpl-bridge/src/lib.rs]: [68dbbfb](https://github.com/futureversecom/trn-seed/pull/884/commits/68dbbfb8c9209a03b2ff03f5b7b3e495bcc57d25)

**Review by JasonTulp — APPROVED** [review]

**Pull request merged**